src/pieces.rs

```diff
@@ -55,3 +55,19 @@ impl AddAssign for Position {
         };
     }
 }
+
+// Add unit tests at the bottom of each file. Each tests module should only have access to super (non integration)
+#[cfg(test)]
+mod tests {
+    use super::Position;
+
+    #[test]
+    fn test_add_position() {
+        let p = Position { x: 1, y: 1 };
+        assert_eq!(p + p, Position { x: 2, y: 2 });
```

Comment:
In the `Move` PR, the `Add` overload actually gets changed to work with a `Position + Move`, so this operation won't be defined.

Reply:
Yeah, it's better to not focus as much on testing until later on when you know you're more confident that you won't change the logic. I just put it here for reference.